scripts/generateDc2InstCat.py

```diff
@@ -0,0 +1,159 @@
+    from lsst.sims.catUtils.baseCatalogModels import GalaxyAgnObj
+    from lsst.sims.utils import _getRotSkyPos
+    import copy
+
+    star_db = StarObj(database='LSSTCATSIM', host='fatboy.phys.washington.edu',
+                      port=1433, driver='mssql+pymssql')
+
+    bulge_db = GalaxyBulgeObj(connection=star_db.connection)
+    disk_db = GalaxyDiskObj(connection=star_db.connection)
+    agn_db = GalaxyAgnObj(connection=star_db.connection)
+
+    if not os.path.exists(out_dir):
+        os.mkdir(out_dir)
+
+    phosim_header_map = copy.deepcopy(DefaultPhoSimHeaderMap)
+    phosim_header_map['nsnap'] = 1
+    phosim_header_map['vistime'] = 30.0
+    phosim_header_map['camconfig'] = 1
+    for obshistid in obshistid_list:
+
+        obs_list = obs_generator.getObservationMetaData(obsHistID=obshistid,
+                                                        boundType='circle',
+                                                        boundLength=args.fov)
+
+        obs = obs_list[0]
+        if dither_switch:
+            print 'dithering'
+            obs.pointingRA = np.degrees(obs.OpsimMetaData['randomDitherFieldPerVisitRA'])
+            obs.pointingDec = np.degrees(obs.OpsimMetaData['randomDitherFieldPerVisitDec'])
+            rotSky = _getRotSkyPos(obs._pointingRA, obs._pointingDec, obs,
+                                   obs.OpsimMetaData['ditheredRotTelPos'])
+
+            obs.rotSkyPos = np.degrees(rotSky)
+            obs.OpsimMetaData['rotTelPos'] = obs.OpsimMetaData['ditheredRotTelPos']
+
+        cat_name = os.path.join(out_dir,'phosim_cat_%d.txt' % obshistid)
+        star_name = 'star_cat_%d.txt' % obshistid
+        gal_name = 'gal_cat_%d.txt' % obshistid
+        agn_name = 'agn_cat_%d.txt' % obshistid
+
+        cat = PhoSimCatalogPoint(star_db, obs_metadata=obs)
+        cat.phoSimHeaderMap = phosim_header_map
+        with open(cat_name, 'w') as output:
+            cat.write_header(output)
+            output.write('includeobj %s.gz\n' % star_name)
+            output.write('includeobj %s.gz\n' % gal_name)
+            output.write('includeobj %s.gz\n' % agn_name)
+
+        star_cat = MaskedPhoSimCatalogPoint(star_db, obs_metadata=obs)
+        star_cat.phoSimHeaderMap = phosim_header_map
+        bright_cat = BrightStarCatalog(star_db, obs_metadata=obs, cannot_be_null=['isBright'])
+        star_cat.min_mag = args.min_mag
+        bright_cat.min_mag = args.min_mag
+
+        from lsst.sims.catalogs.definitions import parallelCatalogWriter
+        cat_dict = {}
+        cat_dict[os.path.join(out_dir, star_name)] = star_cat
+        cat_dict[os.path.join(out_dir, 'bright_stars_%d.txt' % obshistid)] = bright_cat
+        parallelCatalogWriter(cat_dict, chunk_size=100000, write_header=False)
+
+        db_bulge = bulgeDESCQAObject(args.descqa_cat_file)
+        cat = PhoSimDESCQA(db_bulge, obs_metadata=obs, cannot_be_null=['hasBulge'])
+        cat.phoSimHeaderMap = DefaultPhoSimHeaderMap
```

Comment:
since this catalog is being called with `write_header=False`, it should not need a header map